File: src/main/java/com/swp391_09/Koi_BE/repositories/TokenRepository.java
package com.swp391_09.Koi_BE.repositories;

import com.swp391_09.Koi_BE.models.Token;
import com.swp391_09.Koi_BE.models.User;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.List;

public interface TokenRepository extends JpaRepository<Token, Long> {
    List<Token> findByUser(User user);
    Token findByToken(String token);
    Token findByRefreshToken(String token);
}

File: src/main/java/com/swp391/koibe/repositories/TokenRepository.kt
package com.swp391.koibe.repositories

import com.swp391.koibe.models.Token
import com.swp391.koibe.models.User
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.data.jpa.repository.Modifying
import org.springframework.data.jpa.repository.Query
import java.time.LocalDateTime

interface TokenRepository : JpaRepository<Token, Long> {
    fun findByUser(user: User): List<Token>
    fun findByToken(token: String): Token
    fun findByRefreshToken(token: String): Token

    @Modifying
    @Query("UPDATE Token t SET t.expired = true WHERE t.expirationDate < :now AND t.expired = false")
    fun updateExpiredTokens(now: LocalDateTime?)
}

